fastlane/Fastfile

```diff
@@ -0,0 +1,169 @@
+    #todo: update_version_file("../#{project}/Version.h", new_version)
+
+    if UI.confirm "Will commit changes to files"
+      git_commit(path: "#{project}.podspec", message: "Bumped #{project} version to #{new_version}")
+    end
+
+    ## Push the branch
+    if UI.confirm "Will push branch changes to GitHub"
+      push_to_git_remote
+    end
+
+    ## Publish a new release to GitHub.
+    if UI.confirm "Will create a release in GitHub with tag '#{new_tag}' on branch '#{branch}'"
+      github_release = create_github_release("#{project} #{new_version}", new_tag, branch)
+      UI.success "Updated version and created release on Github!"
+      UI.success "Please update its description here: #{github_release['html_url']}"
+      system('open', github_release['html_url'])
+    end
+
+    ## Pull the newly created tag.
+    git_pull(only_tags: true)
+  end
+
+  def lookup(options, key, allowed_values, default=nil)
+    if !options[key]
+      return default
+    else
+      value = options[key]
+      if allowed_values.include? value
+        return value
+      else
+        UI.user_error! "Error! Can't accept value for #{key}. Accepted values are #{allowed_values.join ', '}."
+      end
+    end
+  end
+
+  def new_tag_and_version(previous_version, prefix, version_bump=nil)
+    # Parse the last tag and retrieve each component.
+    major, minor, patch = previous_version.match(/([0-9]+)\.([0-9]+)\.([0-9]+)/).captures
+
+    # Map some of the version components to integers.
+    major = major.to_i
+    minor = minor.to_i
+    patch = patch.to_i
+
+    # Bump version, if needed.
+    # This will reset the release type.
+    case version_bump
+    when "major"
+      major += 1
+      minor = 0
+      patch = 0
+    when "minor"
+      minor += 1
+      patch = 0
+    when "patch"
+      patch += 1
+    end
+
+    return "#{prefix}#{major}.#{minor}.#{patch}", "#{major}.#{minor}.#{patch}"
+  end
+
+  def update_version_file(file, version)
+    open(file, 'w') do |f|
+      f.puts "#define VERSION #{version}"
+    end
+  end
+
+  def verify_branch(branch)
+    # Check if already on branch.
+    current_branch = git_branch
+    if current_branch == branch
+      UI.success "On '#{branch}' branch"
+      return branch
+    else
+      if !UI.confirm "Not on '#{branch}' branch, are you sure you know what you are doing?"
+        UI.user_error! "Not on '#{branch}' branch"
+      end
+      UI.important "On '#{current_branch}' not '#{branch}' branch!\n"
+      return git_branch
+    end
+  end
+
+  def ensure_tag_is_unique(tag)
+    if git_tag_exists(tag: tag, remote: true)
+      UI.user_error! "Tag '#{tag}' already exists! Check github for the full list of tags. Bump the version if necessary."
+    end
+  end
+
+  def create_github_release(title, tag, branch)
+    return set_github_release(
+      repository_name: "finn-no/FinniversKit",
+      name: title,
+      tag_name: tag,
+      api_token: github_token,
```

Comment:
Just noticed this one now. I'm sure we do the same for the app, but what happens if this envvar isn't set before you run the script?

Maybe `verify_env_var` should be called before the `make_release` function within the `make_x_version` lanes? Just so we're sure the envvar is set before we do any work at all. It sucks to do all the steps just to be denied when trying to push the release 😄

Reply:
Yes! Moving it to the very top 👍